tools/build_script_generator/scons/site_tools/artifact.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019, Niklas Hauser
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+import os
+import shutil
+from SCons.Script import *
+import subprocess
+import binascii
+
+def run_store_artifact(target, source, env):
+	artifactpath = os.path.join(env["CONFIG_BUILD_BASE"], "artifacts")
+	try:
+		os.makedirs(artifactpath)
+	except:
+		pass
+	source = str(source[0])
+	binary = os.path.splitext(source)[0]+".bin"
+	subprocess.call(env.subst("$OBJCOPY -O binary {} {}".format(source, binary)), shell=True)
```

Comment:
@dergraaf I need the binary to compute the firmware hash, but I couldn’t figure out how to depend on `env.Bin(source)` properly. Any ideas?

Reply:
In general that's what `env.Depends()` is for. I can look into it in more detail tomorrow, at the moment I don't have my laptop.

Reply:
Yes, I got to `env.Depends` too, but I just didn't understand SCons enough to know *when* to call this? I tried something like `env.Depends(action, "path/to/artifact.bin")` in the `store_artifact` function but I didn't get that to work. SCons is super undocumented.